tree/master/cloud/src/solution/Microsoft.Legal.MatterCenter.Web/Controllers/ConfigController.cs

```diff
@@ -102,7 +105,50 @@ IHostingEnvironment hostingEnvironment
 
                 var configResultsVM = await configRepository.GetConfigurationsAsync(filter);
 
-                CreateConfig(configResultsVM);
+                CreateConfig(configResultsVM, "uiconfig.js", false);
+                return matterCenterServiceFunctions.ServiceResponse(configResultsVM, (int)HttpStatusCode.OK);
+
+            }
+            catch (Exception exception)
+            {
+                customLogger.LogError(exception, MethodBase.GetCurrentMethod().DeclaringType.Name, MethodBase.GetCurrentMethod().Name, logTables.SPOLogTable);
+                throw;
+            }
+        }
+
+
+        /// <summary>
+        /// Returns all the entries for Configuring the UI for sharepoint online
+        /// </summary>
+        /// <param name="filter"></param>
+        /// <returns></returns>
+        [HttpPost("getconfigsforspo")]
+        [SwaggerResponse(HttpStatusCode.OK)]
+        public async Task<IActionResult> GetConfigsForSPO([FromBody] String filter)

```

Comment:
A new web api method to create the ui config js for sharepoint online related pages